modules/server/src/network/aggregations/AggregationAccumulator.ts

```diff
@@ -1,32 +1,62 @@
+import { ALL_NETWORK_AGGREGATION_TYPES_MAP } from '..';
 import { SUPPORTED_AGGREGATIONS } from '../common';
 import { Aggregations, Bucket, NumericAggregations } from '../types/aggregations';
+import { Hits } from '../types/hits';
 import { AllAggregations } from '../types/types';
+import { RequestedFieldsMap } from '../util';
 
 type ResolveAggregationInput = {
-	aggregationsMap: AllAggregations;
+	data: { aggregations: AllAggregations; hits: Hits };
 	accumulator: AllAggregations;
+	requestedFields: string[];
 };
 
 type AggregationsTuple = [Aggregations, Aggregations];
 type NumericAggregationsTuple = [NumericAggregations, NumericAggregations];
 
+const emptyAggregation = (hits: number) => ({
+	bucket_count: 1,
+	buckets: [{ key: '___aggregation_not_available___', doc_count: hits }],
+});
+
+// mutation - update a single aggregations field in the accumulator
+const addToAccumulator = ({ existingAggregation, aggregation, type }) => {
+	// if first aggregation, nothing to resolve with yet
+	return !existingAggregation
+		? aggregation
+		: resolveToNetworkAggregation(type, [aggregation, existingAggregation]);
+};
+
 /**
  * Resolves returned aggregations from network queries into single accumulated aggregation
  *
  * @param
  */
-const resolveAggregations = ({ aggregationsMap, accumulator }: ResolveAggregationInput) => {
-	Object.keys(aggregationsMap).forEach((fieldName) => {
-		const aggregation = aggregationsMap[fieldName];
-		const aggregationType = aggregation?.__typename || '';
-
-		const accumulatedFieldAggregation = accumulator[fieldName];
-
-		// mutation - update a single aggregations field in the accumulator
-		// if first aggregation, nothing to resolve yet
-		accumulator[fieldName] = !accumulatedFieldAggregation
-			? aggregation
-			: resolveToNetworkAggregation(aggregationType, [aggregation, accumulatedFieldAggregation]);
+const resolveAggregations = ({ data, accumulator, requestedFields }: ResolveAggregationInput) => {
+	requestedFields.forEach((requestedField) => {
+		const { aggregations, hits } = data;
+
+		const isFieldAvailable = !!aggregations[requestedField];
+		const type = ALL_NETWORK_AGGREGATION_TYPES_MAP.get(requestedField);
+		const existingAggregation = accumulator[requestedField];
+
+		if (isFieldAvailable) {
+			accumulator[requestedField] = addToAccumulator({
+				existingAggregation,
+				aggregation: aggregations[requestedField],
+				type,
+			});
+		} else {
+			// only need to add empty agg for Aggregations type to account for bucket counts
+			// histogram => buckets is not supported for NumericAggregations
+			if (type === SUPPORTED_AGGREGATIONS.Aggregations) {
+				accumulator[requestedField] = addToAccumulator({
+					existingAggregation,
+					aggregation: emptyAggregation(hits.total),
+					type,
+				});
+			}
+		}
```

Comment:
Adds extra empty bucket (___aggregation_not_available___) to account for fields not available on node:
```javascript
 "has_clinical_data": {
          "bucket_count": 2,
          "buckets": [
            {
              "doc_count": 972,
              "key": "___aggregation_not_available___"
            },
            {
              "doc_count": 1684,
              "key": "__missing__"
            }
          ]
        }
  ```